cpp/ql/src/semmle/code/cpp/rangeanalysis/SignAnalysis.qll

```diff
@@ -207,22 +207,20 @@ private predicate unknownSign(Instruction i) {
 }
 
 /**
- * Holds if `lowerbound` is a lower bound for `bounded` at `pos`. This is restricted
+ * Holds if `lowerbound` is a lower bound for `bounded``. This is restricted
  * to only include bounds for which we might determine a sign.
  */
-private predicate lowerBound(IRGuardCondition comp, Instruction lowerbound, Instruction bounded, Instruction pos, boolean isStrict) {
+private predicate lowerBound(IRGuardCondition comp, Operand lowerbound, Operand bounded, boolean isStrict) {
```

Comment:
To port this library to `Operand`, I thought you'd just change pairs of `(Instruction instr, Instruction pos)` into `Operand op`, but this commit also changes occurrences like `Instruction lowerbound` to have type `Operand` without calling its `.getInstruction()`. That will increase the size of each predicate like this by effectively joining with all the places where `lowerbound` is used. Is there a good reason to do that?

Reply:
This allows us to be more precise when `lowerbound` has a lower bound, as in the following (contrived) example:
```
int f(int x, int y) {
  if (x < 0) {
    return x;
  }
  iif (x < y) {
   return y; // y is strictly positive because of the bound on x above
  }
  return 0;
}```

Reply:
Nice.

Reply:
I've added a test demonstrating this for lower bounds

Reply:
Oh ,there actually is a cartesian product there - I'll see if I can add it in directly or if I need to do a rewrite of `IRGuards.qll` to make it work